src/data/reports/csy-audit-operative/AuditItemD2Repository.ts

```diff
@@ -0,0 +1,381 @@
+                const intraOperative2Ids = this.getColumnValues(intraOperative2Data, etaRegistryId);
+                const intraOperative3Ids = this.getColumnValues(intraOperative3Data, etaRegistryId);
+                const intraOperative4Ids = this.getColumnValues(intraOperative4Data, etaRegistryId);
+                const intraOperative5Ids = this.getColumnValues(intraOperative5Data, etaRegistryId);
+
+                return _.intersection(
+                    intraOperativeIds,
+                    intraOperative2Ids,
+                    intraOperative3Ids,
+                    intraOperative4Ids,
+                    intraOperative5Ids
+                );
+            }
+            case "intraOperativeComplications": {
+                const [
+                    intraOperativeComplicationData,
+                    intraOperativeComplication2Data,
+                    intraOperativeComplication3Data,
+                    intraOperativeComplication4Data,
+                    intraOperativeComplication5Data,
+                    asaScoreData,
+                    medicalComorbiditiesData,
+                ] = data;
+
+                const intraOperativeComplicationIds = this.getColumnValues(
+                    intraOperativeComplicationData,
+                    etaRegistryId
+                );
+                const intraOperativeComplication2Ids = this.getColumnValues(
+                    intraOperativeComplication2Data,
+                    etaRegistryId
+                );
+                const intraOperativeComplication3Ids = this.getColumnValues(
+                    intraOperativeComplication3Data,
+                    etaRegistryId
+                );
+                const intraOperativeComplication4Ids = this.getColumnValues(
+                    intraOperativeComplication4Data,
+                    etaRegistryId
+                );
+                const intraOperativeComplication5Ids = this.getColumnValues(
+                    intraOperativeComplication5Data,
+                    etaRegistryId
+                );
+                const intraOperativeComplicationsMatchedIds = _.union(
+                    intraOperativeComplicationIds,
+                    intraOperativeComplication2Ids,
+                    intraOperativeComplication3Ids,
+                    intraOperativeComplication4Ids,
+                    intraOperativeComplication5Ids
+                );
+
+                const asaScoreIds = this.getColumnValues(asaScoreData, etaRegistryId);
+                const medicalComorbiditiesIds = this.getColumnValues(medicalComorbiditiesData, etaRegistryId);
+                const matchedIds = _.union(asaScoreIds, medicalComorbiditiesIds);
+
+                return _.intersection(intraOperativeComplicationsMatchedIds, matchedIds);
+            }
+            default:
+                return [];
+        }
+    }
+}
+
+const metadata = {
+    programs: {
+        operativeCareProgramId: "Cd144iCAheH",
```

Comment:
I know this could be a lot of more code and work, but normally we prefer to use codes instead hardcoding ids. That way we can prevent future changes if id's are not the same between different environments (local, dev, production, etc).
```ts
const metadata = {
  programs: {
   operativeCareProgramCode: "PROGRAM_CODE"
  }
}

// then we can use the /metadata endpoint to get the details

api.metadata.get({})
```

This is something to discuss with the project manager since this change will not have any impact on the functionality.

Reply:
@eperedo Miquel has suggested that we skip this at the moment. hopefully i will make the changes in a future PR